src/psyclone/tests/domain/lfric/kernel/lfric_kernel_metadata_test.py

```diff
@@ -99,104 +112,79 @@ def test_init_args_error():
-    assert len(tmp.meta_args) == 1
-    assert tmp.meta_args[0] is scalar_arg
-    # TODO issue #1879 gh_shape, meta_funcs, meta_reference_element,
-    # meta_mesh
-
-
-def test_create_from_psyir_error():
-    '''Test that create_from_psyir raises the expected exceptions.'''
-    with pytest.raises(TypeError) as info:
-        _ = LFRicKernelMetadata.create_from_psyir(None)
-    assert ("Expected a DataTypeSymbol but found a NoneType."
-            in str(info.value))
-
-    with pytest.raises(InternalError) as info:
-        _ = LFRicKernelMetadata.create_from_psyir(
-            DataTypeSymbol("x", REAL_TYPE))
-    assert ("Expected kernel metadata to be stored in the PSyIR as an "
-            "UnknownFortranType, but found ScalarType." in str(info.value))
 
 
 METADATA = (
     "type, extends(kernel_type) :: testkern_type\n"
-    "   type(arg_type), dimension(7) :: meta_args =        &\n"
-    "        (/ arg_type(gh_scalar,   gh_real, gh_read),     &\n"
-    "           arg_type(gh_field,    gh_real, gh_inc,  w1), &\n"
-    "           arg_type(gh_field*3,  gh_real, gh_read, w2), &\n"
+    "   type(arg_type), dimension(7) :: meta_args =                       &\n"
+    "        (/ arg_type(gh_scalar,   gh_real, gh_read),                  &\n"
+    "           arg_type(gh_field,    gh_real, gh_inc,  w1),              &\n"
+    "           arg_type(gh_field*3,  gh_real, gh_read, w2),              &\n"
     "           arg_type(gh_field, gh_real, gh_read, w2, "
-    "mesh_arg=gh_coarse), &\n"
+    "mesh_arg=gh_coarse),                                                 &\n"
     "           arg_type(gh_field*3, gh_real, gh_read, w2, "
-    "mesh_arg=gh_fine), &\n"
-    "           arg_type(gh_operator, gh_real, gh_read, w2, w3), &\n"
+    "mesh_arg=gh_fine),                                                   &\n"
+    "           arg_type(gh_operator, gh_real, gh_read, w2, w3),          &\n"
     "           arg_type(gh_columnwise_operator, gh_real, gh_read, w3, "
-    "w0)  &\n"
+    "w0)                                                                  &\n"
     "         /)\n"
+    "   type(func_type), dimension(2) :: meta_funcs =                     &\n"
+    "        (/ func_type(w1, gh_basis),                                  &\n"
+    "           func_type(w2, gh_basis, gh_diff_basis)                    &\n"
+    "        /)\n"
+    "   type(reference_element_data_type), dimension(2) ::                &\n"
+    "     meta_reference_element =                                        &\n"
+    "        (/ reference_element_data_type(normals_to_horizontal_faces), &\n"
+    "           reference_element_data_type(normals_to_vertical_faces)    &\n"
+    "        /)\n"
+    "   type(mesh_data_type) :: meta_mesh(1) =                            &\n"
+    "        (/ mesh_data_type(adjacent_face) /)\n"
+    "   integer :: gh_shape = gh_quadrature_XYoZ\n"
+    "   integer :: gh_evaluator_targets(2) = (/ w0, w3 /)\n"
```

Comment:
This is actually invalid metadata (see https://psyclone.readthedocs.io/en/stable/dynamo0p3.html#gh-shape-and-gh-evaluator-targets). However, you could leave it as is and add a TODO that this will break once we add more correctness checks.

Reply:
Yes, I've not worried about checks between classes in this PR. Added to TODO referencing a new issue #1953.